Library firebase and test connection

FebernalGober opened the pull request:
Install library react-native-firebase and testing connection to Firebase Projects and users accounts

# Description

We install the necessary dependencies to include Firebase in our React Native project and we implement tests in Firebase to analyze the performance

We also edit the corresponding files according to the documentation to link it correctly

Antonio3veces3 committed:
Library firebase and test connection

Install library react-native-firebase and testing connection to Firebase Projects and users accounts

Co-authored-by: FebernalGober <alexis.ferber18@hotmail.com>
Co-authored-by: J3fte <santillanabdiel0@gmail.com>

Tole15 reviewed (commented):
Muy buena prueba chicos.

Antonio3veces3 committed:
duplicate button test

Antonio3veces3 committed:
Add button Login With Google

Install library and button to login with google

Co-authored-by: FebernalGober <alexis.ferber18@hotmail.com>
Co-authored-by: J3fte <santillanabdiel0@gmail.com>

J3fte committed:
Added new components

the components are used for loginScreen was used to access the application, in the SignWithGoogle component a button was added to be able to log in through google, and the last component that was added was the RegisterScreen which by means of an email name and password a user is created whose information is saved in firebase

Co-authored-by: FebernalGober <alexis.ferber18@hotmail.com>
Co-authored-by: Antonio3veces3 <aantonioramirez33@gmail.com>

Antonio3veces3 committed:
Create App navigation, mainly Screens

Create the route and navigation of this App and implement functionality to move from one screen to another.

Co-authored-by: FebernalGober <alexis.ferber18@hotmail.com>
Co-authored-by: J3fte <santillanabdiel0@gmail.com>

J3fte committed:
Added the Queries folder

the components of the functions were separated

Co-authored-by: FebernalGober <alexis.ferber18@hotmail.com>
Co-authored-by: Antonio3veces3 <aantonioramirez33@gmail.com>

J3fte committed:
Merge pull request #4 from BrightCoders-Institute/navigator

Create App navigation and main path

Antonio3veces3 committed:
Resolve conflict

Modify App.tsx to resolve conflict to merge to branch pre-develop

Co-authored-by: FebernalGober <alexis.ferber18@hotmail.com>
Co-authored-by: J3fte <santillanabdiel0@gmail.com>

Antonio3veces3 committed:
Resolving Merge branch pre-develop into authGoogle

Resolving dependencies conflicts to able merge

Co-authored-by: FebernalGober <alexis.ferber18@hotmail.com>
Co-authored-by: J3fte <santillanabdiel0@gmail.com>

Antonio3veces3 committed:
Merge pull request #5 from BrightCoders-Institute/authGoogle

Auth google

Antonio3veces3 committed:
Create initial form to SignUpScreen

Add inputs,labels and checkbox. Add styling to Screen and elements

Co-authored-by: FebernalGober <alexis.ferber18@hotmail.com>
Co-authored-by: J3fte <santillanabdiel0@gmail.com>

J3fte committed:
Persisting authentication state

Use of state of user for obtaining information

Co-authored-by: FebernalGober <alexis.ferber18@hotmail.com>
Co-authored-by: Antonio3veces3 <aantonioramirez33@gmail.com>

Antonio3veces3 committed:
Refactor structure Screen Login

Refactoring structure of Login Screen, simplify components.
Add button SignUp and its style.

Co-authored-by: FebernalGober <alexis.ferber18@hotmail.com>
Co-authored-by: J3fte <santillanabdiel0@gmail.com>

Antonio3veces3 committed:
Input validation

Implement input validation, validate email structure and password,
validate empty inputs and show error message.
When all inputs are correct the buttons to sign up will be enable.

Co-authored-by: FebernalGober <alexis.ferber18@hotmail.com>
Co-authored-by: J3fte <santillanabdiel0@gmail.com>

J3fte committed:
Added context and functions

it was added context the user for it obtain the mail and name, when the
user access in the application

Co-authored-by: FebernalGober <alexis.ferber18@hotmail.com>
Co-authored-by: Antonio3veces3 <aantonioramirez33@gmail.com>

Antonio3veces3 committed:
Implement sign up with firebase

Implement sign up and validation with firebase from the App.

Co-authored-by: FebernalGober <alexis.ferber18@hotmail.com>
Co-authored-by: J3fte <santillanabdiel0@gmail.com>

Antonio3veces3 committed:
Add button Sign up with Google

Add button sign up with google and validate with Firebase Auth
and Firebase Firestore if the use is already sign up or not.

Co-authored-by: FebernalGober <alexis.ferber18@hotmail.com>
Co-authored-by: J3fte <santillanabdiel0@gmail.com>

J3fte committed:
configure the myFlights components

it added a new functionalities in the file myFlights for organizes the app

Co-authored-by: FebernalGober <alexis.ferber18@hotmail.com>
Co-authored-by: Antonio3veces3 <aantonioramirez33@gmail.com>

Antonio3veces3 committed:
Merge pull request #6 from BrightCoders-Institute/loginScreen

Login screen

J3fte committed:
Merge branch 'pre-develop' into myfligthScreen

Co-authored-by: FebernalGober <alexis.ferber18@hotmail.com>
Co-authored-by: Antonio3veces3 <aantonioramirez33@gmail.com>

J3fte committed:
Merge pull request #7 from BrightCoders-Institute/myfligthScreen

Myfligth screen

Antonio3veces3 committed:
Add inputs and style to loginScreen

Antonio3veces3 committed:
Add functionality to button Sign in

Add validation to button Sign in and verify user account

Co-authored-by: FebernalGober <alexis.ferber18@hotmail.com>
Co-authored-by: J3fte <santillanabdiel0@gmail.com>

J3fte committed:
Create styles for My Flights

we create styles for component my flights and separate the functions the file my flights

Co-authored-by: FebernalGober <alexis.ferber18@hotmail.com>
Co-authored-by: Antonio3veces3 <aantonioramirez33@gmail.com>

Antonio3veces3 committed:
Add implementation to Login

Add Google button to login.

Co-authored-by: FebernalGober <alexis.ferber18@hotmail.com>
Co-authored-by: J3fte <santillanabdiel0@gmail.com>

Antonio3veces3 committed:
Deleted button  Google Sign In

Antonio3veces3 committed:
Merge branch 'pre-develop' into myfligthScreen

Antonio3veces3 committed:
Merge pull request #8 from BrightCoders-Institute/myfligthScreen

Myfligth screen

Antonio3veces3 committed:
Add Flight list and get data about user Flights

Co-authored-by: FebernalGober <alexis.ferber18@hotmail.com>
Co-authored-by: J3fte <santillanabdiel0@gmail.com>

FebernalGober committed:
Booking screen with the 5 interior screens

The 5 screens indicated in the documentation were created
- Where are you now?
- Where will you be flying to?
- Select Date
- How Many Passengers?
- Your request was received.
We separate each screen into different components and in all cases we separate the main logic to maintain order in the code structure.
we separate the general styles in a style file, although in the components we decided to put the style right there to avoid a kind of code spaghetti

Co-authored-by: Antonio3veces3 <aantonioramirez33@gmail.com>
Co-authored-by: J3fte <santillanabdiel0@gmail.com>

Antonio3veces3 committed:
Finished MyFlights Screen

Add list of reservations, button to add new reservation and go to
booking.
Get the user reservation from firestore and render in the list items.

Co-authored-by: FebernalGober <alexis.ferber18@hotmail.com>
Co-authored-by: J3fte <santillanabdiel0@gmail.com>

FebernalGober committed:
Code cleanup & scrollToSelecNumber implementation

We removed unnecessary views where we could implement the component itself on all 5 screens
We modified screen 4 so that FlatList would rotate and the selected data would remain in the indicated section

Co-authored-by: Antonio3veces3 <aantonioramirez33@gmail.com>
Co-authored-by: J3fte <santillanabdiel0@gmail.com>

FebernalGober committed:
Change of components to choose passengers

We changed the passenger pick component from a FlatList to a Picker, it had been considered but we noticed documentation about it being deprecated, although the react-native-wheel-pick version was available
we change the component NumberDropDown to PassengersSelector

Co-authored-by: Antonio3veces3 <aantonioramirez33@gmail.com>
Co-authored-by: J3fte <santillanabdiel0@gmail.com>

FebernalGober committed:
Code cleanup, screen line 5

Removed the Navigation function from components that did not need it.
component on screen 5 for aesthetic detail of gray line

Co-authored-by: Antonio3veces3 <aantonioramirez33@gmail.com>
Co-authored-by: J3fte <santillanabdiel0@gmail.com>

Antonio3veces3 committed:
Merge branch 'pre-develop' into booking

Conflict resolved

Antonio3veces3 committed:
Refactor Booking components and get their data

Co-authored-by: FebernalGober <alexis.ferber18@hotmail.com>
Co-authored-by: J3fte <santillanabdiel0@gmail.com>

FebernalGober committed:
"destino" component to update status

We incorporate the destination component so that it is built as the flight data is chosen

Co-authored-by: Antonio3veces3 <aantonioramirez33@gmail.com>
Co-authored-by: J3fte <santillanabdiel0@gmail.com>

Antonio3veces3 committed:
Merge pull request #9 from BrightCoders-Institute/booking

Booking

Antonio3veces3 committed:
Upload new reservation to Firestore

Upload new user reservation to Firestore

Co-authored-by: FebernalGober <alexis.ferber18@hotmail.com>
Co-authored-by: J3fte <santillanabdiel0@gmail.com>

Antonio3veces3 committed:
Solution to bug for render date in the components

Date format resolved and show correctly in the components.

Co-authored-by: FebernalGober <alexis.ferber18@hotmail.com>
Co-authored-by: J3fte <santillanabdiel0@gmail.com>

YilKanda commented on VolcanAirApp/src/Queries/Booking/updateFlighs.tsx:
```diff
@@ -0,0 +1,31 @@
+        });
+      });
+
+      console.log('Flight information updated successfully.');
```
Do not leave debug code on GitHub.

YilKanda commented on VolcanAirApp/src/Queries/Booking/updateFlighs.tsx:
```diff
@@ -0,0 +1,31 @@
+
+      console.log('Flight information updated successfully.');
+    } else {
+      console.log('No user found with the provided email address.');
```
Do not leave debug code on GitHub.

YilKanda commented on VolcanAirApp/src/Queries/Booking/updateFlighs.tsx:
```diff
@@ -0,0 +1,31 @@
+      console.log('No user found with the provided email address.');
+    }
+  } catch (error) {
+    console.error('Error updating flight information:', error);
```
Do not leave debug code on GitHub.

YilKanda commented on VolcanAirApp/src/Queries/LoginQueries.tsx:
```diff
@@ -0,0 +1,16 @@
+    .where('email', '==', email)
+    .get()
+    .then((querySnapshot) => {
+      console.log(querySnapshot.docs);
```
Do not leave debug code on GitHub.

YilKanda commented on VolcanAirApp/src/Queries/LoginQueries.tsx:
```diff
@@ -0,0 +1,16 @@
+      console.log(querySnapshot.docs);
+    })
+    .catch(error => {
+      console.log('Error al realizar la consulta:', error);     
```
Do not leave debug code on GitHub.

YilKanda commented on VolcanAirApp/src/Queries/MyFligths/myFlightsQueries.tsx:
```diff
@@ -0,0 +1,22 @@
+  const data: UserData[] = [];
+  await collectionRef.where('email', '==', email).onSnapshot((querySnapshot) => {
+     setData(querySnapshot.docs[0]._data.flight_reservations);
+    // console.log(querySnapshot.docs[0]._data.flight_reservations);
```
If you don't use this code, delete it.

YilKanda commented on VolcanAirApp/src/Queries/RegisterQueries.tsx:
```diff
@@ -0,0 +1,57 @@
+      }
+    })
+    .catch(error => {
+      console.log('Error al realizar la consulta:', error);     
```
Do not leave debug code on GitHub.

YilKanda commented on VolcanAirApp/src/Queries/RegisterQueries.tsx:
```diff
@@ -0,0 +1,57 @@
+  .createUserWithEmailAndPassword(email, password)
+  .then(() => {
+    register(nombre,email,password, signWithGoogle);
+    console.log('Usuario creado!');
```
Do not leave debug code on GitHub.

YilKanda commented on VolcanAirApp/src/Queries/RegisterQueries.tsx:
```diff
@@ -0,0 +1,57 @@
+  })
+  .catch(error => {
+    if (error.code === 'auth/email-already-in-use') {
+      console.log('El correo ya se esta usando!');
```
Do not leave debug code on GitHub.

YilKanda commented on VolcanAirApp/src/Queries/RegisterQueries.tsx:
```diff
@@ -0,0 +1,57 @@
+    }
+
+    if (error.code === 'auth/invalid-email') {
+      console.log('El correo o la contraseña es invalida!');
```
Do not leave debug code on GitHub.

YilKanda commented on VolcanAirApp/src/Queries/RegisterQueries.tsx:
```diff
@@ -0,0 +1,57 @@
+      console.log('El correo o la contraseña es invalida!');
+    }
+
+    console.error(error);
```
Do not leave debug code on GitHub.

YilKanda commented on VolcanAirApp/src/Queries/RegisterQueries.tsx:
```diff
@@ -0,0 +1,57 @@
+    signWithGoogle: signWithGoogle
+  })
+  .then(() => {
+    console.log('User added!');
```
Do not leave debug code on GitHub.

YilKanda commented on VolcanAirApp/src/Screens/BookingScreen.tsx:
```diff
@@ -0,0 +1,109 @@
+          }
+        </View>
+      </SafeAreaView>
+      /*{currentScreen === 'Screen1' && (
+        <Screen1 navigation={navigation} setCurrentScreen={setCurrentScreen} />
+      )}
+      {currentScreen === 'Screen2' && (
+        <Screen2 navigation={navigation} setCurrentScreen={setCurrentScreen} />
+      )}
+      {currentScreen === 'Screen3' && (
+        <Screen3 navigation={navigation} setCurrentScreen={setCurrentScreen} />
+      )}
+      {currentScreen === 'Screen4' && (
+        <Screen4 navigation={navigation} setCurrentScreen={setCurrentScreen} />
+      )}
+      {currentScreen === 'Screen5' && (
+        <Screen5 navigation={navigation} setCurrentScreen={setCurrentScreen} />
+      )}*/
+    
```
If you don't occupy, delete it.

YilKanda commented on VolcanAirApp/src/Screens/LoginScreen.tsx:
```diff
@@ -0,0 +1,78 @@
+  }, [email, password])
+
+  const inputValidator = ()=>{
+    //validatePassword(password, setPassInputError)
```
If you don't occupy, delete it

YilKanda commented on VolcanAirApp/src/Screens/LoginScreen.tsx:
```diff
@@ -0,0 +1,78 @@
+    try {
+      const userCredential = await auth().signInWithEmailAndPassword(email, password);
+      const user = userCredential.user;  
+      console.log(user);
```
Do not leave debug code on GitHub.

YilKanda reviewed (changes requested):
Do not leave debug code on GitHub. Like the console.log() in your production code.

Check the indentation of your files, so that it is uniform.

If you do not occupy any code, do not comment, delete it.

Remember that on GitHub your code must be clean, easy to read, and functional.
